### PR TITLE
Migrating product editor inventory section to use slot fills

### LINKED ISSUE #36420: [Product Extensibility] Move Inventory fields and section into Slot fill component
**Description**

Similar to what is done in this PR: https://github.com/woocommerce/woocommerce/pull/36368 we want to move the inventory section and field out of the product form and have it added by a Slot fill instead.
We should also add a correct order separated by two from the slot fill above, this will be 1, given its the first item.

This issue should also add the initial slot item for the inventory tab -> `tab/inventory`

**Acceptance criteria**

- [ ] Inventory section and field should be added by the `tab/inventory` section fill.

### PR DESCRIPTION
**Pull request by joelclimbsthings**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Transitioning inventory section in product editor to utilize our slot-fill components. 

**Note:** There were some preexisting tests for the Inventory section. I experimented for a bit with making these work via slot-fill, but it didn't seem to be a simple thing to make work, and I don't see any other instances in the repo of testing slot-filled content. This may need to be a spike or another issue.

Closes #36420 

### How to test the changes in this Pull Request:

1. Checkout branch.
2. Enable product editor at WooCommerce -> Settings -> Advanced -> Features -> Try the new product editor
3. Go to product editor at Products -> Add new (MVP)
4. To general smoke test of inventory section, ensure it behaves exactly the same as previously.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by ROBERLANDIO — APPROVED** [review]
plugins/woocommerce-admin/client/products/fills/inventory-section/inventory-field-stock-manage.tsx

**Review by ROBERLANDIO — COMMENTED** [review]
https://github.com/ROBERLANDIO/-octaedro-approved-this-pull-request.-Nice-job-joshuatf-This-is-testing-well-and-the-code-

File: plugins/woocommerce-admin/client/products/fills/inventory-section/inventory-section-fills.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+					}
+				>
+					<Card>
+						<CardBody>
+							<WooProductFieldItem.Slot
+								section={ INVENTORY_SECTION_ID }
+							/>
+							<CollapsibleContent
+								toggleText={ __( 'Advanced', 'woocommerce' ) }
+							>
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/products/fills/inventory-section/inventory-section-fills.tsx:84]: Nice this seems like a good way to handle it!

**Review by louwie17 — DISMISSED** [review]
Nice work, this tests well!

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 nice work @joelclimbsthings, I did end up rebasing this PR for you.

**Pull request merged**